### PR TITLE
276 - learning elements with the same classification should be displayed in a group besides each other

### PR DESCRIPTION
**Pull request by juli-txt**
## Description

- Added library for automatic import sorting via prettier
- Refactored DropdownLanguage
- Removed not working lock function of ReactFlow Controls
- Added Switch to change between grouped and single view of learning elements
- Added evaluation questionnaire node
- Refactored nodes to remove duplicate code
- Added functionality to group nodes by classification
- Added functionality to automatically sort glossary entries alphabetically

## Type of changes

- [ ] Bugfix 
- [x] Feature 
- [x] Refactoring 
- [ ] Optimization 
- [x] Documentation 
- [ ] CI/CD 
- [ ] Other 

## Requirements
- [x] Everything build out of Material UI components
- [x] New Material UI components capsulated via dependency injection in "@common/components"
- [x] Provide basic information pieces in comments
- [x] Establish interface to backend (if necessary) 
- [x] Use Skeleton components if something has to be fetched (if necessary)
- [x] Error management: Components should handle undefined inputs
- [x] Components should be resuable (if possible)
- [x] [Component Requirements](https://lab.las3.de/nextcloud/index.php/apps/onlyoffice/510074?filePath=%2FHASKI-Extern%2F06-Frontend%2F03-UX%2F04-Implementation%2FComponent_Requirements.docx) are met

### CODE REVIEW

File: src/pages/Topic/Topic.hooks.tsx [hunk 1]
```diff
@@ -85,60 +86,51 @@ export const useTopic = (params?: useTopicHookParams): TopicHookReturn => {
...
   }, [setIsOpen])
 
-  const handleSetUrl = useCallback(
-    (url: string) => {
-      setUrl(url)
-    },
-    [setUrl]
-  )
+  // Groups learning elements by classification.
+  const groupLearningElementsByClassification = useCallback(
```
Comment (DimitriB01) [inline src/pages/Topic/Topic.hooks.tsx:90]: chatti hat mir das ausgespuckt. Finde ich leichter verständlich und scheint auch das selbe Ergebnis zu liefern.   
```
// Groups learning elements by classification.
  const groupLearningElementsByClassification = useCallback(
    (learningPath: LearningPathLearningElement[], nodesGrouped: boolean) => {
      // If nodes are not grouped, return an array of arrays, each containing a single learning element.
      if (!nodesGrouped) {
        return learningPath.map((learningElement) => [learningElement])
      }

      // Group learning elements by classification
      const groupedLearningElements: { [classification: string]: LearningPathLearningElement[] } = {}

      learningPath.forEach((learningPathLearningElement) => {
        const classification = learningPathLearningElement.learning_element.classification
        // Initialize the array for the classification if it doesn't exist
        groupedLearningElements[classification] = groupedLearningElements[classification] || []
        // Push the learning element into the corresponding classification array
        groupedLearningElements[classification].push(learningPathLearningElement)
      })

      // Return an array containing the values of the object, i.e., arrays of learning elements grouped by classification.
      return Object.values(groupedLearningElements);
    },
    []
  )
  ```

File: yarn.lock [hunk 2]
```diff
@@ -95,6 +95,15 @@
     json5 "^2.2.3"
     semver "^6.3.1"
 
+"@babel/generator@7.17.7":
```
Comment (DimitriB01) [inline yarn.lock:98]: normalerweise pushen wir die  yarn.lock ja nicht

**Review by DimitriB01 — COMMENTED** [review]
Nach dem Merge müssen wir Yeimy Bescheid geben, dass der Lernpfad nun immer strikt aufsteigende Lernelemente (position) enthält.
Klassifikationen folgen aufeinander und werden gegrouped.

File: src/pages/Topic/Topic.hooks.tsx (hunk 1 above)
Comment (DimitriB01) [reply src/pages/Topic/Topic.hooks.tsx:90]: Ist wohl hinfällig wegen object-mutation

Comment (juli-txt) [reply src/pages/Topic/Topic.hooks.tsx:90]: The following line contains a mutation:
```
        // Initialize the array for the classification if it doesn't exist
        groupedLearningElements[classification] = groupedLearningElements[classification] || []
```

**Review by DimitriB01 — APPROVED** [review]

**Pull request merged**